lms/djangoapps/courseware/views.py

```diff
@@ -1370,3 +1371,33 @@ def _track_successful_certificate_generation(user_id, course_id):  # pylint: dis
                 }
             }
         )
+
+
+@require_http_methods(["GET", "POST"])
+def render_xblock(request, usage_key_string):
```

Comment:
Have you done sanity checks to make sure the # of queries (both SQL and MongoDB) are roughly the same whether you're loading a subsection via this interface vs. the index page? I'm not looking for full load testing, so much as doing comparisons in DjDT.